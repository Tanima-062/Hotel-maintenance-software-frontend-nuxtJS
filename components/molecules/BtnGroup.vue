<template>
  <div>
    <button
      v-for="(item, index) in items"
      :key="item.value"
      type="button"
      class="border border-solid px-4 py-1.5"
      :class="`${borderRadius(index, items.length)} ${overawedBorder(
        index
      )} ${isActive(item.value === activeValue && isSelect)}`"
      @click="click(item.value)"
    >
      {{ item.name }}
    </button>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'
import { Global } from '~/types'

@Component
export default class extends Vue {
  @Prop({
    default: [
      { name: 'hoge', value: 0 },
      { name: 'hoge1', value: 1 },
      { name: 'hoge2', value: 2 },
    ],
  })
  items!: Global.BtnGroup[]

  @Prop({ default: 0 })
  activeValue!: number

  @Prop({ default: true })
  isSelect!: boolean

  private overawedBorder(index: number): string {
    return index !== 0 ? 'border-l-0' : ''
  }

  private borderRadius(index: number, length: number): string {
    if (index === 0) return 'rounded-l-lg'
    if (index === length - 1) return 'rounded-r-lg'
    return ''
  }

  private isActive(bool: boolean): string {
    return bool
      ? 'bg-btnColor text-white border-btnColor'
      : 'bg-white text-black-light border-btnColor-border'
  }

  @Emit()
  private click(value: number): number {
    return value
  }
}
</script>
