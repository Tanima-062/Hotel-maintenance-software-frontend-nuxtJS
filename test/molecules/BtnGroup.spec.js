import { mount } from '@vue/test-utils'
import BtnGroup from '@/components/molecules/BtnGroup.vue'

describe('Molecules BtnGroup Check', () => {
  test('BtnGroup Test', () => {
    const props = {
      items: [
        { name: 'hoge', value: 0 },
        { name: 'hoge1', value: 1 },
        { name: 'hoge2', value: 2 },
      ],
    }
    const wrapper = mount(BtnGroup, { propsData: props })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
