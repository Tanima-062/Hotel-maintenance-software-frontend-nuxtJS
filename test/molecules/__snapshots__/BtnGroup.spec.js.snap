// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules BtnGroup Check BtnGroup Test 1`] = `
<div>
  <button
    class="border border-solid px-4 py-1.5 rounded-l-lg  bg-btnColor text-white border-btnColor"
    type="button"
  >
    
    hoge
  
  </button>
  <button
    class="border border-solid px-4 py-1.5  border-l-0 bg-white text-black-light border-btnColor-border"
    type="button"
  >
    
    hoge1
  
  </button>
  <button
    class="border border-solid px-4 py-1.5 rounded-r-lg border-l-0 bg-white text-black-light border-btnColor-border"
    type="button"
  >
    
    hoge2
  
  </button>
</div>
`;
